src/pages/contributing.md

```diff
@@ -1,43 +1,200 @@
-We have a wiki elsewhere that can capture more comprehensive code examples.
+```bash
+nvm use # or nvs use on Windows
+npm install
+npm run start
+```
 
-Our audience is anyone who wants to understand how to build a mod using Forge.
+## An Overview
 
-Please don't try to turn this documentation into a tutorial on Java Development - it is intended for people who understand how a Java class works, and other fundamental structures of Java.
+This documentation is a guide to help a modder understand and implement a given concept from Minecraft or NeoForged.
 
-Style Guide
------------
+This documentation is NOT meant as a tutorial, allowing a modder to copy-paste the examples. If you are looking for a tutorial, there are plenty of videos and pages, which are not linked here, that you can use and follow along with.
 
-:::caution
-Please use **two spaces** to indent, not tabs.
-:::
+This documentation is also NOT meant as documentation for a class. Providing a description of an element is unavoidable when writing a guide; however, if you would like to document a class, you should contribute to [Parchment for Minecraft][parchment] or [NeoForge for NeoForged][neo].
+
+Final, this documentation is NOT meant to explain Java concepts. This documentation is intended for people who already has a solid basis in Java. If a Java concept needs to be explained to better understand the concept (such as JVM Descriptors for Access Transformers), a link should be provided to the original resource. Otherwise, if you are unfamiliar with Java, there are plenty of online resources to learn from:
+
+* [JetBrains Academy][jetbrains]
+* [Codeacademy][codeacademy]
+* [University of Helsinki][helsinki]
+* [Oracle][oracle]
+* [Introduction to Programming using Java by David J. Eck][eck]
+
+## Concepts
+
+Each page should guide a modder on a particular concept. If the concept is too large in scope, the concept should be split into separate sub-concepts, each within its own page. For example, if you are writing  a cookbook, there can be a page for each recipe, rather than a single page containing all the recipes.
+
+When describing a concept, you should first introduce what the concept is, where it is used in Minecraft, why it should be used, and how to use it. Each section within a concept should have a header. A section can also be broken into sub-sections if necessary. For example, each recipe within a cookbook can have a sub-section for ingredients and the recipe itself.
+
+If you need to refer to other concepts, the relevant page should be linked along with a summary and/or some example to understand the application.
+
+## Examples
+
+Code examples should generally be pseudocode-like objects meant to enhance the understanding of a modder. When implementing a method, it is usually specific to the desired goal a modder is trying to achieve. As a guide, this documentation aims to be somewhat agnostic to a modder's specific goal, instead covering the general use case.
```

Comment:
This should clarify what "pseudocode-like" means. Generally, pseudocode refers to text which is code-like in structure but is more towards prose-like in syntax. An example pseudocode for a statement like `int x = -1` in Java would be "define x with value of -1".

The example below is more like Java code (in structure and syntax) but with placeholders using comments for specific logic that the developer implements themselves (whether because it's explained by a previous section or because it's specific to their usecase). That is the effect we're aiming for here as I understand it, so it would be nice to clarify that.